### PR TITLE
ai manager for "1822 character interface"

### PR DESCRIPTION
**Pull request by cawke**
https://github.com/WEKIT-ECS/MIRAGE-XR/pull/1911

### CODE REVIEW

**Comment by cawke** [comment]
@AKissMail can you approve it?

File: Assets/Resources/AI_Server.txt
Comment (AKissmehl) [inline Assets/Resources/AI_Server.txt:1]: It is dummy data, but shall we exclude that from the repo or why do we need that?

File: Assets/MirageXR/Scripts/AiServices/AiServices.cs
Comment (AKissmehl) [inline Assets/MirageXR/Scripts/AiServices/AiServices.cs:1]: Ok, this has been moved to [AiServices.cs](https://github.com/WEKIT-ECS/MIRAGE-XR/pull/1974/files/db335a98dbc1847c815cdd493c00166c78afc685#diff-8fadb4998aa96552b96f7bc9d35a8bda9a43733d967d56a1aada56a59e8e744f)?

File: Assets/MirageXR/Managers/AIManager/AiServices.cs [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+            var request = UnityWebRequest.Get(apiURL);
+            request.SetRequestHeader("Authorization", $"Token {token}");
+            await request.SendWebRequest();
+            if (request.result != UnityWebRequest.Result.Success)
+            {
+                throw new HttpRequestException($"Error while receiving the token: {request.error}!");
+            }
+
+            var optionsText = request.downloadHandler.text;
+            var myDeserializedClass = JsonConvert.DeserializeObject<List<OptionsResponse>>(optionsText);
```
Comment (AKissmehl) [inline Assets/MirageXR/Managers/AIManager/AiServices.cs:143]: I had to change here a few things on the server side, we need to update the JSON object here. That is with what I came up with: 
```

[
  {
    "name": "listen/",
    "models": [ "model name as a string array"]
  },
  {
    "name": "speak/",
    "models": [
      {
        "name":"Descriptive model name",
        "Description": "Descriptive of the voice",
        "apiName": "name on the server"
      }
    ]
  },
  {
    "name": "think/",
    "models": [
      {
        "name":"Descriptive model name",
        "Description": "Description of the model properties",
        "apiName": "name on the server"
      }
    ]
  }
]

```

File: Assets/MirageXR/Managers/AIManager/AiServices.cs [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+
+            if (string.IsNullOrEmpty(token))
+            {
+                throw new ArgumentException("token is null");
+            }
+
+            var apiURL = $"{url}/speak/";
+            using var webRequest = UnityWebRequestMultimedia.GetAudioClip(apiURL, AudioType.MPEG);
+            webRequest.SetRequestHeader("Authorization", $"Token {token}");
+            webRequest.SetRequestHeader("message", speakOut);
```
Comment (AKissmehl) [inline Assets/MirageXR/Managers/AIManager/AiServices.cs:186]: That was my weird of a German word.  I would propose to change `message`.

**Review by AKissmehl — APPROVED** [review]

**Pull request merged**